csrc/ir/nodes.cpp

```diff
@@ -4459,14 +4459,16 @@ Val* CatOp::getPred(int input_idx) const {
 
 std::vector<PolymorphicValue> CatOp::evaluate(
     const ExpressionEvaluator& ee,
-    const std::vector<PolymorphicValue>& inputs) const {
-  std::vector<at::Tensor> in;
+    std::unordered_map<const Val*, PolymorphicValue>& known_values) const {
+  // CatOp is preceded by a PadOp internally.
+  // For ATen evaluation, directly compute the unpadded inputs.
+  std::vector<at::Tensor> unpadded_inputs;
   int64_t concat_dim = concatenatedDim();
-  for (auto i : c10::irange(inputs.size())) {
-    auto unpadded_inp = ee.evaluate(input(i)->definition()->input(0));
-    in.push_back(unpadded_inp.as<at::Tensor>());
+  for (auto inp : inputs()) {
+    auto eval_i = ee.evaluate(inp->definition()->input(0), known_values);
```

Comment:
nit: should we `NVF_ERROR` that the `inp->definition()` is a `PadOp`?

Reply:
Yes, would guard against future potential changes.